network/p2p/p2p.go

```diff
@@ -281,9 +280,13 @@ func (s *serviceImpl) dialNode(ctx context.Context, peer *peer.AddrInfo) error {
 
 // AddrInfo returns the peer.AddrInfo for self
 func (s *serviceImpl) AddrInfo() peer.AddrInfo {
+	addrs, err := s.host.Network().InterfaceListenAddresses()
+	if err != nil {
+		s.log.Errorf("failed to get listen addresses: %v", err)
+	}
 	return peer.AddrInfo{
 		ID:    s.host.ID(),
-		Addrs: s.host.(*basichost.BasicHost).AllAddrs(), // fetch all addresses, including private ones
```

Comment:
Are we sure the new call is returning the same addresses/format as this legacy one was?

Reply:
`AllAddrs` is listen addrs + `HasDiscoveredNAT` or `OwnObservedAddrs`

`InterfaceListenAddresses` is only listen addrs.

Our usages are:
```
// p2pNetwork
localAddr, has := n.Address()

// node
func (node *AlgorandFullNode) Start() {
	//...
	node.config.NetAddress, _ = node.net.Address()
	//...
}

func (node *AlgorandFullNode) ListeningAddress() (string, bool) {
	return node.net.Address()
}
```
So we do not expect anything other than a listen address actually, so it fits well.